### PR TITLE
Added cc -Wextra Flag

### LINKED ISSUE #406: Add cc -Wextra flag for extra compile-time checking
Add cc -Wextra for extra compile-time source checking and fix or exclude warnings generated.

### PR DESCRIPTION
**Pull request by danintel**
Added cc -Wextra for extra compile-time source checking.
Fixed or excluded all new warnings.

Fixes #406.

Signed-off-by: Dan Anderson <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by dantpm** [comment]
No there's just spelling errors.  There are a lot of them and I will separate them out.

The only things hiding in the code is bugs we have yet to find.

**Comment by dantpm** [comment]
I separated out the spelling error-only fixes into separate commit.
So there are two commits in this pull request.

File: src/access-broker.c [hunk 1]
```diff
@@ -522,6 +522,7 @@ access_broker_init_tpm (AccessBroker *broker)
 out:
     return rc;
 }
+
```
Comment (flihp) [inline src/access-broker.c]: no superfluous whitespace changes please: https://github.com/tpm2-software/tpm2-abrmd/blob/master/CONTRIBUTING.md#patch-requirements

File: src/access-broker.c [hunk 2]
```diff
@@ -532,7 +533,23 @@ access_broker_get_trans_object_count (AccessBroker *broker,
...
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 } } };
```
Comment (flihp) [inline src/access-broker.c]: This is pretty unwieldy and largely meaningless. I tried enabling `-Wextra` in the past and stuff like this is what got me to back away from the whole thing. Do we have any options to trim this stuff down?

File: src/access-broker.c [hunk 3]
```diff
@@ -667,7 +685,23 @@ access_broker_flush_all_unlocked (AccessBroker     *broker,
...
```
Comment (flihp) [inline src/access-broker.c]: Having this block in two separate tests is begging for at least a macro like the other structure initialization macros we have in the test harness. I think they all have the `STATIC_INIT` suffix.

File: src/random.c [hunk 4]
```diff
@@ -144,10 +144,10 @@ random_get_bytes (Random    *random,
     }
     return i;
 }
+
```
Comment (flihp) [inline src/random.c]: whitespace: https://github.com/tpm2-software/tpm2-abrmd/blob/master/CONTRIBUTING.md#patch-requirements

File: src/random.c [hunk 5]
```diff
@@ -144,10 +144,10 @@ random_get_bytes (Random    *random,
...
 /*
- * Get 64 bits of random data from the parameter 'random' object. If
- * we successfully fill in the supplied 'dest' parameter with random
- * data we return 0. On error return -1.
+ * Get 64 bits of random data from the parameter 'random' object.
+ * On error, return 0 instead of random data.
```
Comment (flihp) [inline src/random.c]: This change isn't mentioned anywhere in the commit message but it doesn't look related to new CFLAGS you've added.

File: src/random.c [hunk 6]
```diff
@@ -160,10 +160,10 @@ random_get_uint64 (Random      *random)
 
     return dest;
 }
+
 /*
- * Get 32 bits of random data from the parameter 'random' object. If
- * we successfully fill in the supplied 'dest' parameter with random
- * data we return 0, on error return -1.
+ * Get 32 bits of random data from the parameter 'random' object.
+ * On error, return 0 instead of random data.
```
Comment (flihp) [inline src/random.c]: same as last comment

File: src/random.c [hunk 7]
```diff
@@ -160,10 +160,10 @@ random_get_uint64 (Random      *random)
 
     return dest;
 }
+
```
Comment (flihp) [inline src/random.c]: whitespace

File: src/random.c [hunk 8]
```diff
@@ -176,6 +176,7 @@ random_get_uint32 (Random       *random)
...
```
Comment (flihp) [inline src/random.c]: whitespace

File: src/tcti-tabrmd.c [hunk 9]
```diff
@@ -433,7 +433,7 @@ typedef struct {
     GBusType type;
 } bus_name_type_entry_t;
 
-const static bus_name_type_entry_t bus_name_type_map[] = {
+static const bus_name_type_entry_t bus_name_type_map[] = {
```
Comment (flihp) [inline src/tcti-tabrmd.c]: This is a good fix. I remember the order of these two qualifiers being unimportant but it looks like C is now obsoleting storage class specifiers anywhere in declarations other than at their beginning.

File: src/tcti-tabrmd.c [hunk 10]
```diff
@@ -641,11 +641,11 @@ Tss2_Tcti_Tabrmd_Init (TSS2_TCTI_CONTEXT *context,
     g_clear_pointer (&conf_copy, g_free);
     g_clear_error (&error);
 
-    return rc;
+    return TSS2_RC_SUCCESS;
```
Comment (flihp) [inline src/tcti-tabrmd.c]: Which of the new flags prompted this change?

File: test/integration/auth-session-start-save.int.c [hunk 11]
```diff
@@ -43,7 +43,7 @@ test_invoke (TSS2_SYS_CONTEXT *sapi_context)
 {
     TSS2_RC               rc;
     TPMI_SH_AUTH_SESSION  session_handle = 0;
-    TPMS_CONTEXT          context = { 0, };
+    TPMS_CONTEXT          context = { 0, 0, 0, { 0, 0 } };
```
Comment (flihp) [inline test/integration/auth-session-start-save.int.c]: Given that this has showed up 4 times now I think it deserves an init macro.

File: test/integration/common.c [hunk 12]
```diff
@@ -41,14 +41,17 @@ create_primary (TSS2_SYS_CONTEXT *sapi_context,
...
+        { 0 } }, { 0, 0 } } };
+    TPM2B_DATA             outside_info    = { 0, 0 };
+    TPML_PCR_SELECTION     creation_pcr    = { 0, { 0, 0, 0 } };
+    TPM2B_PUBLIC           out_public       = { 0, { 0, 0, 0, { 0, 0 }, { 0,
+        { 0 } }, { 0, 0 } } };
+    TPM2B_CREATION_DATA    creation_data   = { 0, { { 0, { 0, 0, 0 } },
+        { 0, 0 }, 0, 0, { 0, 0 }, { 0, 0 }, { 0, 0 } } };
     TPM2B_DIGEST           creation_digest = TPM2B_DIGEST_STATIC_INIT;
-    TPMT_TK_CREATION       creation_ticket = { 0 };
+    TPMT_TK_CREATION       creation_ticket = { 0, 0, { 0, 0 } };
```
Comment (flihp) [inline test/integration/common.c]: init macros please, pretty please 🙏

File: src/access-broker.c (hunk 1 above)
Comment (dantpm) [reply src/access-broker.c]: OK, but there is no mention at the link about whitespace, or at the links at the link.
Just trying to make the code more readable.

File: src/access-broker.c (hunk 3 above)
Comment (dantpm) [reply src/access-broker.c]: OK. I can add a static const or macro for this.

File: src/random.c (hunk 5 above)
Comment (dantpm) [reply src/random.c]: OK. I found the problem by adding the CCFLAG. I'll add a note to the commit message.

File: src/tcti-tabrmd.c (hunk 9 above)
Comment (dantpm) [reply src/tcti-tabrmd.c]: The compiler complained and warned about "const static". Changing to "static const" stopped the warning.

File: src/tcti-tabrmd.c (hunk 10 above)
Comment (dantpm) [reply src/tcti-tabrmd.c]: It should be "return rc". Thanks.

File: test/integration/auth-session-start-save.int.c (hunk 11 above)
Comment (dantpm) [reply test/integration/auth-session-start-save.int.c]: I'll add an init macro, although it really should be in the header file (which is a separate repository).

File: test/integration/common.c (hunk 12 above)
Comment (dantpm) [reply test/integration/common.c]: Again, the init macros belong in a separate repository, but I can add them here in abrmd instead.

File: src/access-broker.c (hunk 2 above)
Comment (dantpm) [reply src/access-broker.c]: I added TPMS_CAPABILITY_DATA_DEFAULT_INIT to the top of the file.

File: src/random.c (hunk 4 above)
Comment (dantpm) [reply src/random.c]: Removed, although it's unclear what the whitespace rule is here. Whitespace between functions makes it more readable.

File: src/random.c (hunk 7 above)
Comment (dantpm) [reply src/random.c]: Removed the blank line to make the code more unreadable :-(.

File: src/random.c (hunk 6 above)
Comment (dantpm) [reply src/random.c]: Reply is same as above--it is mentioned in the commit message and I expanded the text in the commit message.
FYI, the block comment is *NOT* correct and I corrected the comment.
This problem was discovered when I added -Wextra to CCFLAGS.

File: src/random.c (hunk 8 above)
Comment (dantpm) [reply src/random.c]: Done.  Removed line between functions to make it less readable.

**Comment by dantpm** [comment]
Merged with upstream changes.  All issues addressed. Thanks for the review!

File: src/random.c (hunk 6 above)
Comment (flihp) [reply src/random.c]: Gotcha, hard to return -1 when you're returning an unsigned number :\

File: Makefile.am [hunk 13]
```diff
@@ -13,7 +13,7 @@ AM_CFLAGS = $(EXTRA_CFLAGS) \
     $(TSS2_SYS_CFLAGS) $(TCTI_DEVICE_CFLAGS) $(TCTI_SOCKET_CFLAGS) \
     $(CODE_COVERAGE_CFLAGS)
 AM_LDFLAGS = $(EXTRA_LDFLAGS) $(CODE_COVERAGE_LIBS)
-UNIT_AM_CFLAGS = $(AM_CFLAGS) $(CMOCKA_CFLAGS)
+UNIT_AM_CFLAGS = $(AM_CFLAGS) $(CMOCKA_CFLAGS) -Wno-unused-parameter
```
[the remark this answers is not in the document]
Comment (dantpm) [reply Makefile.am]: I added it specifically to the files where there are functions with unused parameters.  Usually because of a fixed prototype.
An alternative is to disable it globally.

Per our chat session, I will add a macro to the functions where it appears to force the parameter to be used.  Something like:
#define UNUSED_PARAM(p) (void)(p);

**Comment by dantpm** [comment]
I addressed the latest issue, disabling unused parameter warnings with #define UNUSED_PARAM(p), and all earlier issues.

**Comment by dantpm** [comment]
Updated newest code, integrated after this pull request, to also pass cc -Wextra checks.

This should be ready to integrate.
